### PR TITLE
Shard lower

### PR DESCRIPTION
**Pull request by JasonTheMain**
## Changes proposed in this pull request:
- Fixing for chance of race condition, now only creates when alias doesn't exist
-
-

## Security considerations

None

### CODE REVIEW

File: jobs/upload_opensearch_config/templates/bin/upload-config.erb [hunk 1]
```diff
@@ -116,5 +116,8 @@ echo "--- License"
...
-curl -XPUT ${CA} ${CERT} ${KEY} ${BASEURL}/%3Clogs-app-%7Bnow%2Fd%7D-01%3E -H 'Content-Type: application/json' -d'{"aliases": {"'${ALIAS}'": {}}}'
\ No newline at end of file
+HTTP_CODE=$(curl -s -o /dev/null -s -w "%{http_code}\n" -I  ${CA} ${CERT} ${KEY} ${BASEURL}/_alias/${ALIAS}/ )
+if [ "$HTTP_CODE" -ge "201" ]; then
+  curl -XPUT ${CA} ${CERT} ${KEY} ${BASEURL}/_plugins/_ism/policies/logs-app-rollover_policy -H 'Content-Type: application/json' -d'{"policy":{"description":"Rollover to a new index when age or size becomes our target","default_state":"Rollover","states":[{"name":"Rollover","actions":[{"retry":{"count":3,"backoff":"exponential","delay":"1m"},"rollover":{"min_index_age":"'${INDEX_AGE}'","min_primary_shard_size":"'${DOCUMENT_SIZE}'","copy_alias":false}}],"transitions":[]}],"ism_template":[{"index_patterns":["logs-app-*"],"priority":100}]}}'
+  curl -XPUT ${CA} ${CERT} ${KEY} ${BASEURL}/%3Clogs-app-%7Bnow%2Fd%7D-01%3E -H 'Content-Type: application/json' -d'{"aliases": {"'${ALIAS}'": {}}}'
```
Comment (markdboyd) [inline jobs/upload_opensearch_config/templates/bin/upload-config.erb:122]:
```suggestion
  curl -XPUT ${CA} ${CERT} ${KEY} ${BASEURL}/${ALIAS} -H 'Content-Type: application/json' -d'{"aliases": {"'${ALIAS}'": {}}}'
```

Comment (JasonTheMain) [reply jobs/upload_opensearch_config/templates/bin/upload-config.erb:122]: That is the index not the Alias

File: jobs/upload_opensearch_config/templates/bin/upload-config.erb [hunk 2]
```diff
@@ -116,5 +116,8 @@ echo "--- License"
...
-curl -XPUT ${CA} ${CERT} ${KEY} ${BASEURL}/%3Clogs-app-%7Bnow%2Fd%7D-01%3E -H 'Content-Type: application/json' -d'{"aliases": {"'${ALIAS}'": {}}}'
\ No newline at end of file
+HTTP_CODE=$(curl -s -o /dev/null -s -w "%{http_code}\n" -I  ${CA} ${CERT} ${KEY} ${BASEURL}/_alias/${ALIAS}/ )
+if [ "$HTTP_CODE" -ge "201" ]; then
+  curl -XPUT ${CA} ${CERT} ${KEY} ${BASEURL}/_plugins/_ism/policies/logs-app-rollover_policy -H 'Content-Type: application/json' -d'{"policy":{"description":"Rollover to a new index when age or size becomes our target","default_state":"Rollover","states":[{"name":"Rollover","actions":[{"retry":{"count":3,"backoff":"exponential","delay":"1m"},"rollover":{"min_index_age":"'${INDEX_AGE}'","min_primary_shard_size":"'${DOCUMENT_SIZE}'","copy_alias":false}}],"transitions":[]}],"ism_template":[{"index_patterns":["logs-app-*"],"priority":100}]}}'
```
[the remark this answers is not in the document]
Comment (JasonTheMain) [reply jobs/upload_opensearch_config/templates/bin/upload-config.erb]: I moved it so the policy is outside of the conditonal

File: jobs/upload_opensearch_config/templates/bin/upload-config.erb (hunk 1 above)
Comment (markdboyd) [reply jobs/upload_opensearch_config/templates/bin/upload-config.erb:122]: Ah, okay

**Review by markdboyd — APPROVED** [review]

**Pull request merged**